File: src/main/java/com/sinnerschrader/s2b/accounttool/logic/exception/BusinessException.java
package com.sinnerschrader.s2b.accounttool.logic.exception;

/** */
public class BusinessException extends Exception {

	private final String code;

	private final Object[] args;

	public BusinessException(String msg, String code) {
		this(msg, code, (Object[]) null);
	}

	public BusinessException(String msg, String code, Object[] args) {
		super(msg);
		this.args = args;
		this.code = code;
	}

	public BusinessException(String msg, String code, Throwable t) {
		this(msg, code, null, t);
	}

	public BusinessException(String msg, String code, Object[] args, Throwable t) {
		super(msg, t);
		this.args = args;
		this.code = code;
	}

	public String getCode() {
		return code;
	}

	public Object[] getArgs() {
		return args;
	}
}

File: src/main/kotlin/com/sinnerschrader/s2b/accounttool/logic/exception/BusinessException.kt
package com.sinnerschrader.s2b.accounttool.logic.exception

@Deprecated("remove")
class BusinessException : Exception {
    val code: String
    val args: Array<Any>?

    @JvmOverloads constructor(msg: String, code: String, args: Array<Any>? = null) : super(msg) {
        this.args = args
        this.code = code
    }

    constructor(msg: String, code: String, t: Throwable) : this(msg, code, null, t)

    constructor(msg: String, code: String, args: Array<Any>?, t: Throwable) : super(msg, t) {
        this.args = args
        this.code = code
    }
}
